src/dropdown/dropdown.scss

```diff
@@ -231,7 +237,10 @@ $height-small: 30px !default;
 
   &.has-value .Select-control {
     padding-right: 45px;
-    color: $text-color-has-value;
+
+    .Select-value .Select-value-label {
+      color: $text-color-has-value;
```

Comment:
here you're finally using `text-color-has-value` but it's working only because you added also `.Select-control` otherwise it would be overwritten by the first one